routes/students.routes.js

```diff
@@ -0,0 +1,62 @@
+const router = require('express').Router();
+const User = require('../models/User.model');
+const {
+  isLoggedIn,
+  checkId,
+  checkRoles,
+  checkIfStudentOrPM,
+} = require('./../middleware');
+
+router.get('/', isLoggedIn, (req, res, next) => {
+  User.find({ role: 'STUDENT' })
+    .then(user =>
+      res.render('students/students-list', {
+        user,
+        isLogged: req.session.currentUser,
+        isPM: req.session.currentUser?.role === 'PM',
+      })
+    )
+    .catch(err => console.log(err));
+});
+
+router.get('/:studentId', isLoggedIn, (req, res, next) => {
+  const { studentId } = req.params;
+
+  User.findById(studentId)
+    .then(student =>
+      res.render('students/student-profile', {
+        student: student,
+        isLogged: req.session.currentUser,
+        isPM: req.session.currentUser?.role === 'PM',
+      })
+    )
+    .catch(err => console.error(err));
+});
```

Comment:
Para la iteración 4 había dos piezas necesarias. Por un lado enseñarle los botones al dueño del perfil

```suggestion
router.get('/:studentId', isLoggedIn, (req, res, next) => {
  const { studentId } = req.params;

  User.findById(studentId)
    .then(student =>
      res.render('students/student-profile', {
        student: student,
        isLogged: req.session.currentUser,
        isPM: req.session.currentUser?.role === 'PM',
        isOwnProfile: req.session.currentUser?.role === studentId
      })
    )
    .catch(err => console.error(err));
});
```

Y ese valor añadido usarlo en la vista de hbs para mostrar los botones.